src/XCCDF/xccdf_session.c

```diff
@@ -491,6 +491,39 @@ static inline int _xccdf_session_load_xccdf_benchmark(struct xccdf_session *sess
 	return 0;
 }
 
+static int _acquire_xccdf_checklist_from_tailoring(struct xccdf_session* session)
+{
+	struct ds_sds_session *ds_sds_session = xccdf_session_get_ds_sds_session(session);
+	if (ds_sds_session == NULL) {
+		return 1;
+	}
+
+	xmlDoc *tailoring_xmlDoc = xmlCopyDoc(oscap_source_get_xmlDoc(session->xccdf.source), true);
+	struct oscap_source *tailoring_source = oscap_source_new_from_xmlDoc(tailoring_xmlDoc, NULL);
+	struct xccdf_tailoring* tailoring = xccdf_tailoring_import_source(tailoring_source, NULL);
+	if (tailoring == NULL) {
+		return 1;
+	}
+	const char *benchmark_ref = oscap_strdup(xccdf_tailoring_get_benchmark_ref(tailoring));
```

Comment:
OK. I will fix that. I need it, because otherwise it would produce a horrible bunch of segfaults.